### ISSUE TITLE
Architecture issue (network morphism) when running on multiple GPUs

### ISSUE THREAD

**Issue by jasoniswow** [keras-team/autokeras#1136]
**Short summary about the issue/question**:
When running example with multiple GPUs, the best architecture (with highest accuracy) is just a simple expansion of original network, wider and deeper, without complex subnet based on skip operations. 

**Brief what process you are following**: 
https://github.com/microsoft/nni/blob/master/examples/trials/network_morphism/cifar10/cifar10_keras.py

**How to reproduce it**: 

1. When running on 1 GPU, the final best model is based on various operation including wider, deeper and skip. 
2. When running on 2 GPUs (1 trial on each GPU), the final best model is only based on wider & deeper operation, without skip operation (subnet). 

**Debugged**:
1. both local and remote mode have this issue.
2. multiple versions including 1.0, 1.1, 1.5, have this issue. 
3. the observation is based on repeated experiments on 1 or 2 GPUs. 

***The understanding of the model generation process is based on nni/networkmorphism_tuner/bayesian.py***:
1. in network morphism, the generation loop will generate multiple "temp_graphs", each corresponding to variable "t". The bigger "t" is, the sooner the generation of the graph is.
2. calculate acquisition function value (acq, the predicted accuracy) for each "temp_graph"
3. choose the "temp_graph" with highest "acq" as the "target_graph". If multiple, choose the one with largest "t".
4. the "target_graph" is the model to be trained.

**To-be-debugged**:
Ran on both 1 GPU and 2 GPUs, we exported the "bayesian.generate" generated "temp_graph" and final "target_graph", then realized the final chosen "temp_graph" is inconsistent with the "target_graph" if 2 GPUs, while being consistent on 1 GPU.

**nni Environment**: 
- nni version: 1.0, 1.1 and 1.5
- nni mode(local|pai|remote): local and remote
- OS: ubuntu 16.04
- python version: 3.7
- is conda or virtualenv used?: conda
- is running in docker?: yes

**need to update document(yes/no)**: no

**Anything else we need to know**: no
